Nanoleaf-wpf/Api/Orchestrator.cs

```diff
@@ -0,0 +1,143 @@
+            _scheduleTimer = new ScheduleTimer(this);
+            _events = new Events.Events(this);
+
+            if (device.OperationMode == OperationMode.Schedule)
+            {
+                _scheduleTimer.StartTimer();
+            }
+            else if (device.OperationMode == OperationMode.Manual)
+            {
+                Task.Run(() => ActivateEffect(Device.OverrideEffect, Device.OverrideBrightness));
+            }
+        }
+
+        public async Task<bool> TrySetOperationMode(OperationMode operationMode, bool isFromOverride = false)
+        {
+            if (!isFromOverride && Device.OperationMode == OperationMode.Manual)
+            { //Only the override is able to stop the override, effects and events may not remove the manual mode
+                return false;
+            }
+
+            Device.OperationMode = operationMode;
+
+            //Stop all things that can activate an effect
+            _scheduleTimer.StopTimer();
+            _events.StopAllEvents();
+
+            if (_customEffects.HasActiveEffects())
+            {
+                await _customEffects.DeactivateAllEffects();
+            }
+
+            //If its a schedule, then the schedule timer can start again. The events and override manage their own effect activation
+            if (Device.OperationMode == OperationMode.Schedule)
+            {
+                _scheduleTimer.StartTimer();
+            }
+
+            return true;
+        }
+
+        public async Task ActivateEffect(string effectName, int brightness)
+        {
+            try
+            {
+                var client = NanoleafClient.GetClientForDevice(Device);
+
+                //DO NOT change the order of disabling effects, then setting brightness and then enabling effects
+                if (_customEffects.HasActiveEffects(effectName))
+                {
+                    await _customEffects.DeactivateAllEffects();
+                }
+
+                await client.StateEndpoint.SetBrightnessAsync(brightness);
+
+                if (_customEffects.EffectIsCustomEffect(effectName))
+                {
+                    var customEffect = _customEffects.GetCustomEffect(effectName);
+
+                    if (!customEffect.IsActive())
+                    {
+                        await customEffect.Activate();
+                    }
+                }
+                else
+                {
+                    await client.EffectsEndpoint.SetSelectedEffectAsync(effectName);
+                }
+
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e, $"Enabling effect failed for device {Device.Name} with trigger effect {effectName}");
+            }
+        }
+
+        public List<Effect> GetCustomEffectAsEffects()
+        {
+            return _customEffects.GetCustomEffectAsEffects();
+        }
+
+        #region Static methods
+        public static void Initialize()
```

Comment:
Orchestrator class has 2 purposes, manage the Device - Orchestrator relationship and the Orchestrator logic.
It seems if the static methods were moved to a class managing the relationship it will be single purpose design.

Reply:
Nice one, is fixed now